solution/runner.go

```diff
@@ -0,0 +1,183 @@
+package solution
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/mini-maxit/worker/executor"
+	"github.com/mini-maxit/worker/utils"
+	"github.com/mini-maxit/worker/verifier"
+)
+
+type LanguageType int
+
+const (
+	PYTHON LanguageType = iota
+	CPP
+)
+
+type SolutionRunner interface {
+	RunSolution() SolutionResult
+}
+
+type SolutionResult struct {
+	Success     bool         // wether solution was without any error
+	StatusCode  int          // any status code in case of error or success
+	Code        string       // any code in case of error or success
+	Message     string       // any information message in case of error is error message
+	TestResults []TestResult // test results in case of error or success
+}
+type TestResult struct {
+	InputFile    string // Path to the input file used for the test
+	ExpectedFile string // Path to the expected output file
+	ActualFile   string // Path to the actual output file produced by the solution
+	Passed       bool   // Whether the test passed or failed
+	ErrorMessage string // Error message in case of failure (if any)
+}
+
+type Runner struct {
+}
+
+type LanguageConfig struct {
+	Type    LanguageType // Which type of language (cpp, c, py)
+	Version string       // Version should be valid version supported by the executor
+}
+
+// TODO: replace ALL Fatalf with approproiate SolutionResult, to inform end user about the error instead of crushing like oioioi))))
+// Input path contains points to directory with input files. File name should be of format {number}.in
+//
+// language is the language of the solution
+//
+// baseDir is the base directory where solution file is stored together with compiled file if applicable
+//
+// solutionFileName is the name of the solution file
+//
+// inputDir is the directory containing input files inside baseDir
+//
+// outputDir is the directory where correct output files are stored
+func (r *Runner) RunSolution(language LanguageConfig, baseDir string, solutionFileName string, inputDir string, outputDir string) SolutionResult {
+
+	// Init appropriate executor
+	var exec executor.Executor
+	var err error
+	switch language.Type {
+	// case PYTHON:
+	// 	exec, err = executor.NewPyExecutor(language.Version)
+	case CPP:
+		exec, err = executor.NewCppExecutor(language.Version)
```

Comment:
I guess we are not considering memory and time limits for now?

Reply:
We should, I just have not come with how to pass them effectively